Ensure that flag_arrays are boolean

bhazelton opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
- Added booleans to parameter types and fixed equality checking for arrays with strict types.
- Added tests that integer flag arrays fail checks.
- Some test fixture cleanup in test_uvdata.py

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. If this PR closes an issue, put the word 'closes' before the issue link to auto-close the issue when the PR is merged. -->
fixes #1039 

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation change (documentation changes only)
- [ ] Version change
- [ ] Build or continuous integration change


## Checklist:
<!--- You may remove the checklists that don't apply to your change type(s) or just leave them empty -->
<!--- Go over all the following points, and replace the space with an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have read the [contribution guide](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/main/.github/CONTRIBUTING.md).
- [x] My code follows the code style of this project.

Bug fix checklist:
- [x] My fix includes a new test that breaks as a result of the bug (if possible).
- [x] All new and existing tests pass.
- [x] I have updated the [CHANGELOG](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/main/CHANGELOG.md).

mkolopanis commented on pyuvdata/parameter.py:
```diff
@@ -192,6 +176,34 @@ def __eq__(self, other):
+                    # types must match in the other direction
+                    if not isinstance(other.value.item(0), self.expected_type):
+                        print(
+                            f"{self.name} parameter has incompatible dtypes. Left is "
+                            f"{other.value.dtype}, right is {other.expected_type}"
```
We're using `other` for both left and right here. Should right be `self.expected_type`?

mkolopanis commented on pyuvdata/parameter.py:
```diff
@@ -225,6 +237,24 @@ def __eq__(self, other):
+                    # types must match in the other direction
+                    if not isinstance(other.value, self.expected_type):
+                        print(
+                            f"{self.name} parameter has incompatible types. Left is "
+                            f"{type(other.value)}, right is {other.expected_type}"
```
same comment here about `other` being both left and right.

bhazelton replied:
Yes, I fixed this one and the one before to be more clear.

mkolopanis reviewed (commented):
This looks really good. Thanks for cleaning up the tests as well. I have a question about some docstrings but should be good to go after that.

bhazelton commented on pyuvdata/uvdata/tests/test_uvdata.py:
```diff
@@ -255,6 +255,28 @@ def hera_uvh5_split(hera_uvh5_split_main):
+    hera_uvh5_xx = UVData()
+    hera_uvh5_xx.read_uvh5(os.path.join(DATA_PATH, "zen.2457698.40355.xx.HH.uvcA.uvh5"))
+
+    return hera_uvh5_xx
```
me neither, happy to add some cleanup.

bhazelton commented:
@mkolopanis I think I addressed your comments, but now the linter and doctest are failing on the initial linux spinup. I noticed that those two jobs have the os specified to be "linux" whereas the regular tests use a docker image.

mkolopanis commented:
Interesting that we specifically set the OS there, I would think we would be agnostic to it. Maybe they are older copy/pasta?

mkolopanis commented on pyuvdata/parameter.py:
```diff
@@ -225,6 +238,25 @@ def __eq__(self, other):
+                        print(
+                            f"{self.name} parameter has incompatible types. Left "
+                            f"requires {type(self.value)}, right is "
+                            "{other.expected_type}"
```
looks like we're missing an `f` here?
```suggestion
                            f"{other.expected_type}"
```

bhazelton commented:
@mkolopanis I fixed the CI issues on this PR
